### PR TITLE
curl -J: overwrite the destination file

### LINKED ISSUE #3380: `curl -JO` appends data to an existing output file
### I did this
```
curl -JO --location https://github.com/curl/curl/releases/download/curl-7_63_0/curl-7.63.0.tar.xz.asc
curl -JO --location https://github.com/curl/curl/releases/download/curl-7_63_0/curl-7.63.0.tar.xz.asc
```

### I expected the following
I expected `curl-7.63.0.tar.xz.asc` to be written only once.  However, it was written twice and concatenated together.  Note that it is not needed to use `--location` to trigger the bug.  It looks like the fix in #2813 was incomplete.  Originally reported at https://bugzilla.redhat.com/1658574

### curl/libcurl version
`curl-7_63_0-13-g9e6518481`

[curl -V output]
```
curl 7.63.0-DEV (x86_64-pc-linux-gnu) libcurl/7.63.0-DEV OpenSSL/1.1.1 zlib/1.2.11 brotli/1.0.7 libidn2/2.0.5 libpsl/0.20.2 (+libidn2/2.0.5) nghttp2/1.35.1
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz brotli TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL 
```

### operating system
Linux

### PR DESCRIPTION
**Pull request by bagder**
Fixes #3380

### CODE REVIEW

File: src/tool_cb_wrt.c [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 /* create a local file for writing, return TRUE on success */
 bool tool_create_output_file(struct OutStruct *outs,
-                             bool append)
+                             bool allow_overwrite)
```
Comment (danielgustafsson) [inline src/tool_cb_wrt.c]: The prototype in `src/tool_cb_wrt.h` needs to be updated to match to avoid confusion.

Comment (bagder) [reply src/tool_cb_wrt.c]: good point!

**Comment by kdudka** [comment]
With this fix applied, curl overwrites the existing local file, instead of appending it.  Is it correct?

As I understand it, the `--remote-header-name` option was originally designed such that it does not overwrite existing local files based on the response header...

**Comment by bagder** [comment]
Oh right, I forgot about that detail. Thanks @kdudka, I'll adapt.

**Comment by danielgustafsson** [comment]
> On 17 Dec 2018, at 15:21, Kamil Dudka <notifications@github.com> wrote:

> With this fix applied, curl overwrites the existing local file, instead of appending it. Is it correct?
> 
> As I understand it, the --remote-header-name option was originally designed such that it does not overwrite existing local files based on the response header…

Good catch, I was confusing myself over the interaction between -O and -J, but you are absolutely correct.

**Comment by kdudka** [comment]
It works as expected now.  Thanks!

**Pull request closed without merging**